x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/rule_diff_tab_react_diff_viewer_continued.tsx

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useContext } from 'react';
+import ReactDiffViewer, { DiffMethod } from 'react-diff-viewer-continued';
+import { get } from 'lodash';
+import {
+  EuiSpacer,
+  EuiAccordion,
+  EuiTitle,
+  EuiFlexGroup,
+  EuiSwitch,
+  EuiHorizontalRule,
+  EuiRadioGroup,
+  useGeneratedHtmlId,
+  useEuiTheme,
+} from '@elastic/eui';
+import type { RuleFieldsDiff } from '../../../../../common/api/detection_engine/prebuilt_rules/model/diff/rule_diff/rule_diff';
+import type { RuleResponse } from '../../../../../common/api/detection_engine/model/rule_schema/rule_schemas.gen';
+
+const HIDDEN_FIELDS = ['meta', 'rule_schedule', 'version'];
+
+const CustomStylesContext = React.createContext({});
+const CompareMethodContext = React.createContext(DiffMethod.CHARS);
+
+interface FieldsProps {
+  fields: Partial<RuleFieldsDiff>;
+  openSections: Record<string, boolean>;
+  toggleSection: (sectionName: string) => void;
+}
+
+const Fields = ({ fields, openSections, toggleSection }: FieldsProps) => {
+  const styles = useContext(CustomStylesContext);
+  const compareMethod = useContext(CompareMethodContext);
+
+  const visibleFields = Object.keys(fields).filter(
+    (fieldName) => !HIDDEN_FIELDS.includes(fieldName)
+  );
+
+  return (
+    <>
+      {visibleFields.map((fieldName) => {
+        const currentVersion: string = get(fields, [fieldName, 'current_version'], '');
+        const mergedVersion: string = get(fields, [fieldName, 'merged_version'], '');
+
+        const oldSource =
+          compareMethod === DiffMethod.JSON && typeof currentVersion === 'object'
```

Comment:
`currentVersion` will always be a string so this always evaulates to `JSON.stringify(currentVersion, null, 2)`. Same in the calculation of `newSource`